### ISSUE TITLE
Place itemgroups like furniture

### ISSUE THREAD

**Issue by BorkBorkGoesTheCode** [CleverRaven/Cataclysm-DDA#19316]
This may have been suggested somewhere before:

Could the mapgen blueprint be used to place itemgroup spawns like it currently places furniture and terrain?

**Comment by John-Candlebury** [comment]
I'm pretty sure this can already be done. I think mapgen-test.json has an example.

**Comment by BorkBorkGoesTheCode** [comment]
I found it. It cannot be used to place an arbitrary number of itemgroups on or in a table or display cases.

**Comment by Coolthulhu** [comment]
What do you mean? Show us some example in "pseudojson" of what exactly do you want to do here. It's hard to express it all when we have 3 different methods of spawning things, all of which are described using similar words.

**Comment by BorkBorkGoesTheCode** [comment]
Like this:

At location "a" on the mapgen blueprint ( "a" could be any ASCII character.), spawn "item_group": "foo" x times out of 100, "item_group": "bar" x times out of 100, etc..

**Comment by Coolthulhu** [comment]
Could probably be hacked around by wrapping them in a new item group and using that.

**Comment by BorkBorkGoesTheCode** [comment]
I found the JSON specification in mapgen.md and mapgen-test.json. It's supposed to work like:

```
"place_groups": {
  "T": { "item": "hive",    "chance": 70 )
}
```

I tried to add a version of it to a custom mapgen, but it didn't work.

**Comment by pisskop** [comment]
Boop - plop

,
            "mapping": {
                "B": { "items": [ { "item": "novels", "chance": 50 }, { "item": "textbooks", "chance": 25 } ] },
                "b": { "items": { "item": "bed", "chance": 60 } },
                "D": { "items": { "item": "dresser", "chance": 75 } },
                "d": { "items": { "item":  "NC_HUNTER_misc", "chance": 50 } },
                "s": { "items": { "item": "traveler", "chance": 15 } },
                "C": { "items": [ { "item": "kitchen", "chance": 50 }, { "item": "fridge", "chance": 50 } ] },
                "X": { "items": [ { "item": "hardware", "chance": 30 }, { "item": "mischw", "chance": 30 } ] },
                "x": { "items": [ { "item": "cleaning", "chance": 30 }, { "item": "hardware_plumbing", "chance": 30 } ] }
            }


Hows that for examples?

**Comment by pisskop** [comment]
<"place_groups": {
  "T": { "item": "hive",    "chance": 70 )
}>

^^^ that doesnt do anything because your brackets are crossed

**Comment by BorkBorkGoesTheCode** [comment]
Thank you! I'm going to go through the mapgen files and convert them to this system as needed.

**Comment by egomassive** [comment]
Something is going haywire with this system. I've tried it with "items"
`"items": {
  "T": { "item": "hive", "chance": 70 }
}`
and with "mapping"
`"mapping": {
  "T": { "Items": { "item": "hive", "chance": 70 } }
}`
I would expect this to produce a single item from the item_group hive with 70% chance. Instead I typically get 3 items. If I bump the chance up to 80 I typically get 4. If I set the chance at 100 I always get one item. Omitting the chance declaration gives 0 items.
